Skip broadcast buffers for SyncBN in DDP

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46636 Skip broadcast buffers for SyncBN in DDP**

Differential Revision: [D24457141](https://our.internmc.facebook.com/intern/diff/D24457141)

mrshenli committed:
Skip broadcast buffers for SyncBN in DDP

[ghstack-poisoned]

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"




[ghstack-poisoned]

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"




[ghstack-poisoned]

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"


Differential Revision: [D24457141](https://our.internmc.facebook.com/intern/diff/D24457141)

[ghstack-poisoned]

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"


Differential Revision: [D24457141](https://our.internmc.facebook.com/intern/diff/D24457141)

[ghstack-poisoned]

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"


Differential Revision: [D24457141](https://our.internmc.facebook.com/intern/diff/D24457141)

[ghstack-poisoned]

rohan-varma commented on torch/testing/_internal/distributed/distributed_test.py:
```diff
@@ -2692,6 +2692,15 @@ def _test_DistributedDataParallel_SyncBatchNorm(self, gpu_subset, rank, local_bs
+
+            # calculate total buffer size from all SyncBN layers
+            syncbn_buffer_size = 0
+            for m_name, m_layer in BN_NET.named_modules():
```
nit: `_` instead of `m_name` since it is not used

mrshenli committed:
Update on "Skip broadcast buffers for SyncBN in DDP"


Differential Revision: [D24457141](https://our.internmc.facebook.com/intern/diff/D24457141)

[ghstack-poisoned]
